examples/sphinx_gallery/benchmark_healpix_pad_timing.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+"""
+Benchmark different HEALPix padding implementations
+---------------------------------------------------
+"""
+import time
+
+import torch
+
+from earth2grid import healpix
+from earth2grid.healpix import pad_backend
+
+# Print GPU information
+if torch.cuda.is_available():
+    print("CUDA available: Yes")
+    print(f"Current device: {torch.cuda.current_device()}")
+    print(f"Device name: {torch.cuda.get_device_name()}")
+    print(f"Device count: {torch.cuda.device_count()}")
+    print(f"Device capability: {torch.cuda.get_device_capability()}")
+else:
+    print("CUDA not available")
+print("\n")
+
+nside = 128
+
+
+neval = 10
+
+
+def test_func(label, pad, compile=False):
+    # warm up
+    if compile:
+        pad = torch.compile(pad)
```

Comment:
Is there any situation where someone might not want to run with torch.compile? If not, might be better to just decorate the function with `@torch.compile` so it's on by default?